File: bot.py
import asyncio
import logging
from aiogram import Bot, Dispatcher, types, F
from aiogram.filters import Command
from aiogram.types import InlineKeyboardMarkup, InlineKeyboardButton, ReplyKeyboardMarkup, KeyboardButton
from aiogram.fsm.storage.memory import MemoryStorage
from config import BOT_TOKEN, CREATOR_ID
from database import Database

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

bot = Bot(token=BOT_TOKEN)
dp = Dispatcher(storage=MemoryStorage())
db = Database()


def get_main_keyboard(user_id: int) -> ReplyKeyboardMarkup:
    buttons = [[KeyboardButton(text="📨 Послать запрос на совместный блог")]]
    if user_id == CREATOR_ID:
        buttons.append([KeyboardButton(text="📋 Посмотреть запросы")])
    return ReplyKeyboardMarkup(keyboard=buttons, resize_keyboard=True)


@dp.message(Command("start"))
async def cmd_start(message: types.Message):
    await message.answer(
        "👋 Добро пожаловать в <b>писмо хандеру</b>!\n\n"
        "Хочешь вести совместный блог? Отправь запрос — и я передам его создателю.\n\n"
        "Выбери действие ниже ",
        parse_mode="HTML",
        reply_markup=get_main_keyboard(message.from_user.id)
    )


@dp.message(F.text == "📨 Послать запрос на совместный блог")
async def send_request(message: types.Message):
    user = message.from_user

    # Collect user info
    username = f"@{user.username}" if user.username else "нет username"
    full_name = user.full_name or "нет имени"
    user_id = user.id

    # Save request to DB
    db.add_request(user_id=user_id, full_name=full_name, username=username)

    # Notify creator
    try:
        await bot.send_message(
            chat_id=CREATOR_ID,
            text=(
                f"🔔 <b>Новый запрос на совместный блог!</b>\n\n"
                f"👤 Имя: {full_name}\n"
                f"🔗 Username: {username}\n"
                f"🆔 ID: <code>{user_id}</code>\n\n"
                f"<a href='tg://user?id={user_id}'>📩 Написать напрямую</a>"
            ),
            parse_mode="HTML"
        )
    except Exception as e:
        logger.warning(f"Не удалось уведомить создателя: {e}")

    await message.answer(
        "✅ <b>Запрос отправлен!</b>\n\n"
        "Создатель блога получил твои контактные данные и скоро свяжется с тобой. 🎉",
        parse_mode="HTML"
    )


@dp.message(F.text == "📋 Посмотреть запросы")
async def view_requests(message: types.Message):
    if message.from_user.id != CREATOR_ID:
        await message.answer("⛔ У тебя нет доступа к этой функции.")
        return

    requests = db.get_all_requests()

    if not requests:
        await message.answer("📭 Запросов пока нет.")
        return

    text = f"📋 <b>Все запросы ({len(requests)}):</b>\n\n"
    for i, req in enumerate(requests, 1):
        text += (
            f"{i}. 👤 <b>{req['full_name']}</b>\n"
            f"   🔗 {req['username']}\n"
            f"   🆔 <code>{req['user_id']}</code>\n"
            f"   <a href='tg://user?id={req['user_id']}'>📩 Написать</a>\n"
            f"   📅 {req['date']}\n\n"
        )

    # Split if too long
    if len(text) > 4000:
        chunks = [text[i:i+4000] for i in range(0, len(text), 4000)]
        for chunk in chunks:
            await message.answer(chunk, parse_mode="HTML", disable_web_page_preview=True)
    else:
        await message.answer(text, parse_mode="HTML", disable_web_page_preview=True)


async def main():
    logger.info("Бот запущен!")
    await dp.start_polling(bot)


if __name__ == "__main__":
    asyncio.run(main())

File: supabase/functions/telegram-bot/bot.ts
import { Bot } from "https://esm.sh/grammy@1.34.0";
import { addRequest, getAllRequests } from "./db.ts";

const BOT_TOKEN = Deno.env.get("BOT_TOKEN")!;
const CREATOR_ID = parseInt(Deno.env.get("CREATOR_ID")!);

export const bot = new Bot(BOT_TOKEN);

function mainKeyboard(userId: number) {
  const keyboard = [["📨 Послать запрос на совместный блог"]];
  if (userId === CREATOR_ID) keyboard.push(["📋 Посмотреть запросы"]);
  return { keyboard, resize_keyboard: true };
}

function formatDate(iso: string): string {
  const d = new Date(iso);
  const pad = (n: number) => n.toString().padStart(2, "0");
  return `${pad(d.getDate())}.${pad(d.getMonth() + 1)}.${d.getFullYear()} ${pad(d.getHours())}:${pad(d.getMinutes())}`;
}

bot.command("start", async (ctx) => {
  await ctx.reply(
    "👋 Добро пожаловать в <b>писмо хандеру</b>!\n\nХочешь вести совместный блог? Отправь запрос — и я передам его создателю.\n\nВыбери действие ниже",
    { parse_mode: "HTML", reply_markup: mainKeyboard(ctx.from!.id) }
  );
});

bot.hears("📨 Послать запрос на совместный блог", async (ctx) => {
  const user = ctx.from!;
  const username = user.username ? `@${user.username}` : "нет username";
  const fullName = [user.first_name, user.last_name].filter(Boolean).join(" ") || "нет имени";

  await addRequest(user.id, fullName, username);

  try {
    await ctx.api.sendMessage(
      CREATOR_ID,
      `🔔 <b>Новый запрос на совместный блог!</b>\n\n👤 Имя: ${fullName}\n🔗 Username: ${username}\n🆔 ID: <code>${user.id}</code>\n\n<a href='tg://user?id=${user.id}'>📩 Написать напрямую</a>`,
      { parse_mode: "HTML" }
    );
  } catch (e) {
    console.warn("Не удалось уведомить создателя:", e);
  }

  await ctx.reply(
    "✅ <b>Запрос отправлен!</b>\n\nСоздатель блога получил твои контактные данные и скоро свяжется с тобой. 🎉",
    { parse_mode: "HTML" }
  );
});

bot.hears("📋 Посмотреть запросы", async (ctx) => {
  if (ctx.from!.id !== CREATOR_ID) {
    await ctx.reply("⛔ У тебя нет доступа к этой функции.");
    return;
  }

  const requests = await getAllRequests();

  if (!requests.length) {
    await ctx.reply("📭 Запросов пока нет.");
    return;
  }

  let text = `📋 <b>Все запросы (${requests.length}):</b>\n\n`;
  for (const [i, req] of requests.entries()) {
    text += `${i + 1}. 👤 <b>${req.full_name}</b>\n   🔗 ${req.username}\n   🆔 <code>${req.user_id}</code>\n   <a href='tg://user?id=${req.user_id}'>📩 Написать</a>\n   📅 ${formatDate(req.created_at)}\n\n`;
  }

  for (let i = 0; i < text.length; i += 4000) {
    await ctx.reply(text.slice(i, i + 4000), { parse_mode: "HTML", disable_web_page_preview: true });
  }
});
